spark-on-watson-studio/nested-json-to-table.md

```diff
@@ -0,0 +1,64 @@
+# Extract and save nested JSON elements into separate Db2 tables
+This example extracts nested elements from a JSON array and saves each array item into new Db2 tables.  
+
+**Note:** this is an introductory example and does not include any Db2 primary key-foreign key table relationships.
+
+The example below will use this JSON document:
+```
+{
+  "_id": "90174e7b66274b00222ef6d8e60807f4",
+  "array_of_objs": [
+    {
+      "number": 100,
+      "boolean": true,
+      "nestedKey": "value"
+    },
+    {
+      "number": 200,
+      "boolean": false,
+      "nestedKey": "nextValue"
+    }
+  ]
+}
+```
+
+## Python example
+```python
+# Import and initialize the SparkSession
+from pyspark.sql import SparkSession
+spark = SparkSession.builder.getOrCreate()
+
+# Load Cloudant documents into a Spark DataFrame from the specified account
+cloudantdata = spark.read.format("org.apache.bahir.cloudant")\
+.option("cloudant.host", "account.cloudant.com")\
+.option("cloudant.username", "username")\
+.option("cloudant.password", "password")\
+.load("database")
+
+# Calculate the number of elements in the 'array_of_objs' column
+nElements = len(cloudantdata.select('array_of_objs').collect()[0].array_of_objs)
+
+# Create a new DataFrame for 'array_of_objs'
+array_of_objs = cloudantdata.select('array_of_objs')
+
+# Create properties for saving the DataFrame to a Db2 table
+properties = {
+    'user': 'username',
+    'password': 'password',
+    'driver': 'com.ibm.db2.jcc.DB2Driver'
+}
+db2_jdbc_url = 'jdbc:db2://****:50000/BLUDB'
+
+# Create a new Spark DataFrame for each nested object in the 'array_of_objs' array
+for i in range(nElements):
+    # New dataframe for i item in 'array_of_objs' array
+    new_table = array_of_objs.withColumn('array_of_objs-' + str(i), cloudantdata.array_of_objs.getItem(i))
+    # Extract each nested object into their own column
+    new_table = new_table.withColumn('array_of_objs-' + str(i) + '.number', new_table['array_of_objs-' + str(i)].number)\
```

Comment:
does this really need a re-assignment to `new_table`? it's already a long line, it might be a bit simpler to read without it

Reply:
As discussed lack of inline comments means this is a no-go, so ignore